packages/x-components/src/x-modules/next-queries/components/next-queries-list.vue

```diff
@@ -17,184 +17,219 @@
-          }
-          return items;
-        },
-        [...this.injectedListItems]
-      );
-    }
-
-    /**
-     * Checks if the next queries are outdated taking into account the injected query.
-     *
-     * @returns True if the next queries are outdated, false if not.
-     * @internal
-     */
-    protected get nextQueriesAreOutdated(): boolean {
-      return (
-        !!this.injectedQuery &&
-        (this.$x.query.nextQueries !== this.injectedQuery ||
-          this.$x.status.nextQueries !== 'success')
-      );
-    }
-
-    /**
-     * Checks if the number of items is smaller than the offset so a group
-     * should be added to the injected items list.
-     *
-     * @returns True if a group should be added, false if not.
-     * @internal
-     */
-    protected get hasNotEnoughListItems(): boolean {
-      return (
-        !this.showOnlyAfterOffset &&
-        !this.hasMoreItems &&
-        this.injectedListItems !== undefined &&
-        this.injectedListItems.length > 0 &&
-        this.offset > this.injectedListItems.length
-      );
+    },
+    setup(props, { slots }) {
+      useRegisterXModule(nextQueriesXModule);
+
+      const $x = use$x();
+
+      const renderSlots = slots;
+
+      /**
+       * The state next queries.
+       *
+       * @internal
+       */
+      const nextQueries: ComputedRef<NextQuery[]> = useGetter('nextQueries', [
+        'nextQueries'
+      ]).nextQueries;
+
+      /**
+       * Injected query, updated when the related request(s) have succeeded.
+       */
+      const injectedQuery = inject<Ref<string | undefined>>(QUERY_KEY as string);
+
+      /**
+       * Indicates if there are more available results than the injected.
+       */
+      const hasMoreItems = inject<Ref<boolean>>(HAS_MORE_ITEMS_KEY as string);
+
+      /**
+       * The grouped next queries based on the given config.
+       *
+       * @returns A list of next queries groups.
+       * @internal
+       */
+      const nextQueriesGroups = computed((): NextQueriesGroup[] => {
+        return Object.values(
+          groupItemsBy(nextQueries?.value, (_, index) =>
+            Math.floor(index / props.maxNextQueriesPerGroup)
+          )
+        )
+          .slice(0, props.maxGroups)
+          .map(nextQueries => ({
+            modelName: 'NextQueriesGroup' as const,
+            id: nextQueries.map(nextQuery => nextQuery.query).join(','),
+            nextQueries
+          }));
+      });
+
+      /**
+       * It injects {@link ListItem} provided by an ancestor as injectedListItems.
+       *
+       * @internal
+       */
+      const injectedListItems = inject<Ref<ListItem[]>>(LIST_ITEMS_KEY as string);
+
+      /**
+       * Checks if the next queries are outdated taking into account the injected query.
+       *
+       * @returns True if the next queries are outdated, false if not.
+       * @internal
+       */
+      const nextQueriesAreOutdated = computed((): boolean => {
```

Comment:
The returned type is already inferred by TS in the most of `computed`. You can avoid the return type